addons/udes_stock/models/stock_picking.py

```diff
@@ -1551,20 +1555,33 @@ def _swap_moveline_info_at_pack_level(self, pack, values):
         )
         return values
 
-    def _swap_moveline_info_below_pack_level(self, move_lines_to_use, pack):
+    def _swap_moveline_info_below_pack_level(
+        self, move_lines_to_use, pack, fields_to_consider=None, fields_to_update=None
+    ):
         """
         Swapping of fields which are more granular than pack level - from quant to move_lines
         specifically, the lots.
         """
+
+        if fields_to_update is None:
+            fields_to_update = {"lot_id"}
+        if fields_to_consider is None:
+            fields_to_consider = {"product_id"}
+
         MoveLine = self.env["stock.move.line"]
         move_lines_to_skip = MoveLine.browse()
-        for (product, lot), quant_ids in pack.mapped("quant_ids").groupby(
-            lambda quant: (quant.product_id, quant.lot_id)
+        for fields, quant_ids in pack.mapped("quant_ids").groupby(
+            lambda quant: {
+                field: getattr(quant, field) for field in fields_to_consider.union(fields_to_update)
+            },
+            sort=False,
```

Comment:
Group by without sorting cannot guarantee to group OK, elements must be sorted first.